### ISSUE TITLE
ImportError: No module named objectid when run on Mac OS X 

### ISSUE THREAD

**Issue by els-pnw** [calmrat/metrique#114]
Python version 2.7.2 on Mac OS X 
Metrique version 0.1.5-23

Run from ipython notebook, the following ImportError occurs.

```
ImportError                               Traceback (most recent call last)
<ipython-input-1-cb06b4bd1bbe> in <module>()
----> 1 from metrique import pyclient
       2 from metrique.container import Container
/Users/esammons/build/bdist.macosx-10.8-intel/egg/metrique/__init__.py in <module>()
/Users/esammons/build/bdist.macosx-10.8-intel/egg/metrique/utils.py in <module>()
ImportError: No module named objectid
```

When run from the python interactive shell, simply as import metrique.

```
>>> import metrique
Traceback (most recent call last):
File "<stdin>", line 1, in <module>
File "build/bdist.macosx-10.8-intel/egg/metrique/__init__.py", line 33, in <module>    
File "build/bdist.macosx-10.8-intel/egg/metrique/utils.py", line 6, in <module>
ImportError: No module named objectid
>>> 
```

**Comment by els-pnw** [comment]
Author: Eric L. Sammons elsammons@gmail.com
Date:   Wed Oct 16 08:29:38 2013 -0400

```
Update for Issue #114, the bson.objectid import ObjectId code does
not work with older versions of pymongo.
```

```
diff --git a/src/metriqueu/metriqueu/utils.py b/src/metriqueu/metriqueu/utils.py
index 2208855..c49bde8 100644
--- a/src/metriqueu/metriqueu/utils.py
+++ b/src/metriqueu/metriqueu/utils.py
@@ -2,7 +2,13 @@
 # vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
 # Author: "Chris Ward" <cward@redhat.com>

-from bson.objectid import ObjectId
+import pkg_resources
+
+if pkg_resources.get_distribution("pymongo").version >= '2.6.3':
+    from bson.objectid import ObjectId
+else:
+    print "You must update your pymongo."
+    sys.exit(-1)
 from calendar import timegm
 from datetime import datetime
 from dateutil.parser import parse as dt_parse
```

**Comment by els-pnw** [comment]
Another way to fix the issue, probably better as we don't waste cycles checking the version and we reduce number of libraries imported.

```
commit 97430f84fedf26958056d6c2bbbf0354cf60381a
Author: Eric L. Sammons <elsammons@gmail.com>
Date:   Wed Oct 16 08:52:23 2013 -0400

    Updated fix for issue #114 to use try/except.

diff --git a/src/metriqueu/metriqueu/utils.py b/src/metriqueu/metriqueu/utils.py
index c49bde8..f044a61 100644
--- a/src/metriqueu/metriqueu/utils.py
+++ b/src/metriqueu/metriqueu/utils.py
@@ -2,13 +2,15 @@
 # vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
 # Author: "Chris Ward" <cward@redhat.com>

-import pkg_resources
-
-if pkg_resources.get_distribution("pymongo").version >= '2.6.3':
+try:
     from bson.objectid import ObjectId
-else:
-    print "You must update your pymongo."
-    sys.exit(-1)
+except ImportError, ie:
+    try:
+        from pymongo.objectid import ObjectId
+    except ImportError, ie:
+        print "Failed to load objectid, %s" % ie
+        sys.exit(-1)
+
 from calendar import timegm
 from datetime import datetime
 from dateutil.parser import parse as dt_parse
```
